torch/nn/parallel/distributed.py

```diff
@@ -654,6 +657,30 @@ def _register_comm_hook(self, state: object, hook: callable):
             >>>     fut.set_result(bucket.get_tensors())
             >>>     return fut
 
+        Example::
+            Below is an example of a simple allreduce hook.
+
+            >>> ddp._register_comm_hook(state = None, hook = allreduce)
+
+            >>> def allreduce(state: object, bucket: dist.GradBucket): -> torch.futures.Future
```

Comment:
hmm, the return type is not `torch.futures.Future`, as they are different, right?

Actually what does `type(work.get_future())` return?

Reply:
It returns `torch._C.Future`